server/services/spotify-service.js

```diff
@@ -38,11 +38,12 @@ module.exports = class SpotifyService {
     })
   }
 
-  getPlaylists () {
-    // this.setCredentials()
-    return spotifyApi.getMe().then(data => {
-      console.log('getMe() returned', data.body)
-      return spotifyApi.getUserPlaylists(data.body.id, {limit: 50}).then(data => data.body)
-    })
+  async getPlaylists () {
+    const user = await spotifyApi.getMe().then(data => data.body)
+    const sampling = await spotifyApi.getUserPlaylists(user.id, {limit: 1}).then(data => data.body)
+    const pageCount = Math.ceil(sampling.total / limit)
+    const pages = Array.from(Array(pageCount), (_, x) => x)
+    const promises = pages.map(p => spotifyApi.getUserPlaylists(user.id, {limit, offset: p * limit}).then(data => data.body.items))
+    return Promise.all(promises).then(results => results.reduce((acc, val) => [...acc, ...val]))
```

Comment:
Isn't `[...acc, ...val]` here basically `acc.concat(val)`? Or am I missing something?

Reply:
Yes it is. I'll make that switch. Spread operator was being too fancy here.